sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RequestHandler.java

```diff
@@ -40,21 +40,27 @@
   public static Response handleMainOperation(Callable<Response> action) {
     try {
       return action.call();
-    } catch (StatusRuntimeException grpcE) { // from gRPC
-      return failResponseForGrpcException(grpcE);
-    } catch (NotFoundException nfe) { // does this actually occur?
+    } catch (Exception e) {
+      return failResponseForAnyException(e);
+    }
+  }
+
+  public static Response failResponseForAnyException(Exception e) {
```

Comment:
would this me not needed once the #1474 is merged?

Reply:
Yes, possibly; I haven't looked into that yet. I think there are trade-offs between local and global handling but I think you are right, for "anything else" cases global handler is probably better.

Reply:
Yes, there's a non trivial conflict; need to resolve post-merge.